File: Server/three-d-portfolio-server/src/main/java/org/threeDPortfolioGallery/workloads/Exhibition.java
package org.threeDPortfolioGallery.workloads;

import com.fasterxml.jackson.annotation.JsonIgnore;
import io.quarkus.hibernate.orm.panache.PanacheEntity;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import javax.persistence.*;
import java.util.List;

@Entity
public class Exhibition extends PanacheEntity {

    public String thumbnail_url;

    public String title;

    // relationship

    @OneToMany(mappedBy = "exhibition")
    public List<Exhibit> exhibits;

    @ManyToOne
    public Theme theme;

    @JsonIgnore
    // TODO: check if okay like that, Cascade types
    @ManyToOne(cascade = CascadeType.ALL )
    public User user;

    @OneToMany(mappedBy = "exhibition")
    public List<Room> rooms;

    @ManyToMany(mappedBy = "exhibitions")
    List<Category> categories;
}

File: Gallery/src/app/shared/class/exhibition.ts
import {Exhibit} from "./exhibit";
import {Room} from "./room";
import {Tag} from "./tag";

export class Exhibition {
  id: number;
  thumbnail_url : string | undefined;
  title: string;
  room_id: number;
  description : string | undefined;
  exhibits : Exhibit[] | undefined;
  room : Room | undefined;
  tags : Tag[] | undefined;

  constructor(id: number, thumbnail_url: string | undefined, title: string, room_id: number, description: string | undefined, exhibits: Exhibit[] | undefined, room: Room | undefined, tag: Tag[] | undefined) {
    this.id = id;
    this.thumbnail_url = thumbnail_url;
    this.title = title;
    this.room_id = room_id;
    this.description = description;
    this.exhibits = exhibits;
    this.room = room;
    this.tags = tag;
  }
}
